app/controllers/user_batches_controller.rb

```diff
@@ -14,11 +14,22 @@ def new
     render("form")
   end
 
+  def upload
```

Comment:
Should there be a manual authorization here?  i.e. something like `authorize!(:create, UserBatch)`